### PR TITLE
Feature/Add Mapkit annotation page and bug demo

### PR DESCRIPTION
**Pull request by DechengMa**
This PR is to add a new map page with annotations, with offset bug that can be easily debugged by `Reveal`

## Changes
- Create `presentModally` to `SceneCoordinator` and `SceneTransitionType` for pushing mapkit modally.
- Create `PhotoAnnotation` class, `BaseAnnotationView` for rendering annotation.
- Create `MapViewController`, `MapViewModel` with `MapViewController.xib` for holding mapview 
- Add a map button in `PhotoDetailsViewController` for navigating to mapview

#### 1. There will be a map icon on the bottom of photo detail controller when a photo has location data. 
image
#### 3. A call out view will be presented when annotation view get clicked
image

#### 3. Gif
Screen Recording 2020-09-04 at 4 13 44 pm

## How to debug the Offset problem in Reveal
As we can see, the pin actual location on the map is the **centre** of the pin view, not the **bottom**, so when you zoom in the pin will fly around (not flying, but you will get the point after watching the below demo).
Screen Recording 2020-09-04 at 4 18 22 pm

This can be debugged and adjust by Reveal with no effort, after giving it an offset of `-20Y` 
image
Problem solved! Now feel confident to change the offset of the annotation to -20 with the folloing code
```
annotationView?.centerOffset = CGPoint(x: 0, y: -20)
```

### CODE REVIEW

File: Papr/Assets.xcassets/map-pin-icon.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    {
+      "idiom" : "universal",
+      "scale" : "1x"
+    },
+    {
+      "idiom" : "universal",
+      "scale" : "2x"
+    },
+    {
+      "filename" : "defualtMarker2.png",
```
Comment (tonyarnold) [inline Papr/Assets.xcassets/map-pin-icon.imageset/Contents.json]: This filename has a typo - `defualt` → `default`

File: Papr/Scenes/Map View/Annotation/BaseAnnotationView.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    func renderSelfAsImage() -> UIImage {
+        let renderer = UIGraphicsImageRenderer(bounds: self.bounds)
+        return renderer.image { rendererContext in
+            self.layer.render(in: rendererContext.cgContext)
+        }
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
```
Comment (tonyarnold) [inline Papr/Scenes/Map View/Annotation/BaseAnnotationView.swift]: FYI, you can more reusably express unimplemented methods like so:

```suggestion
        fatalError("\(#function) has not been implemented")
```

File: Podfile.lock [hunk 3]
```diff
@@ -58,9 +58,9 @@ SPEC CHECKSUMS:
   RxNuke: 7c56f40f26bc3ec715cff5f5355119ca7989b010
   RxRelay: 89d54507f4fd4d969e6ec1d4bd7f3673640b4640
   RxSwift: e2dc62b366a3adf6a0be44ba9f405efd4c94e0c4
-  TinyNetworking: f373ddd43edc5320ff98059ff484591970fbfb17
+  TinyNetworking: c0b2cbfe8db8451e42c3371a3eecfc61538b310a
```
Comment (tonyarnold) [inline Podfile.lock:65]: Was there a reason that you updated the TinyNetworking dependency?

**Review by tonyarnold — DISMISSED** [review]
The source looks good to me - I've left a few comments inline, but use your judgement as to whether you want to use the suggestions or not. 

I do think we need to change the callout image. We can't use the likeness/name of anyone without their permission. Perhaps use one of those online AI face generators to make something (and a name) that's completely fictional?

File: Papr/Assets.xcassets/map-pin-icon.imageset/Contents.json (hunk 1 above)
Comment (DechengMa) [reply Papr/Assets.xcassets/map-pin-icon.imageset/Contents.json]: Oh, I didn't see this, nice catch!

File: Podfile.lock (hunk 3 above)
Comment (DechengMa) [reply Podfile.lock:65]: This is an update I didn't expect, after running `pod install` this line changed. Will reset this one.

Comment (DechengMa) [reply Podfile.lock:65]: Sorry I installed another package for generating random names so this can't be reverted 😢

Comment (tonyarnold) [reply Podfile.lock:65]: That's fine - something look out for in the future. 

Generally, I try to stay on top of updates to dependencies, even without a reason for it, but I do like to sequester them into their own PR when I make the change, just so any breakage can be isolated.

**Review by tonyarnold — APPROVED** [review]
Seems solid - thanks for addressing the issues, Decheng!

**Pull request merged**